extensions/ql-vscode/src/remote-queries/view/RawResultsTable.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { Box, Link } from '@primer/react';
+import { CellValue, RawResultSet, ResultSetSchema } from '../../pure/bqrs-cli-types';
+import { useState } from 'react';
+import TextButton from './TextButton';
+
+const numOfResultsInContractedMode = 5;
+
+const Row = ({
+  row
+}: {
+  row: CellValue[]
+}) => (
+  <>
+    {row.map((cell, cellIndex) => (
+      <Box key={cellIndex}
+        borderColor="border.default"
+        borderStyle="solid"
+        justifyContent="center"
+        alignItems="center"
+        p={2}>
+        <Cell value={cell} />
+      </Box>
+    ))}
+  </>
+);
+
+const Cell = ({
+  value
+}: {
+  value: CellValue
+}) => {
+  switch (typeof value) {
+    case 'string':
+    case 'number':
+    case 'boolean':
+      return <span>{value.toString()}</span>;
+    case 'object':
+      // TODO: This will be converted to a proper link once there
+      // is support for populating link URLs.
+      return <Link>{value.label}</Link>;
```

Comment:
At some point (not now), you will have to make sure that non-printable characters and whitespace are properly handled. Some queries specifically look at whitespace. There was a query that looks for zero-width spaces (since they are often an indication of some funny business). We needed to specifically handle this in for local query results viewing.

Reply:
Ohh thanks for injecting your domain knowledge, super useful! I'll take a look.